boom/bootloader.py

```
@@ -2465,12 +2465,12 @@ def write_entry(self, force: bool = False, expand: bool = False):
             chmod(entry_path, BOOT_ENTRY_MODE)
-        except Exception as e:
-            _log_error("Error writing entry file %s: %s", entry_path, e)
+        except (OSError, IOError) as err:  # pragma: no cover
+            _log_error("Error writing entry file %s: %s", entry_path, err)
             try:
                 unlink(tmp_path)
-            except Exception:
-                _log_error("Error unlinking temporary path %s", tmp_path)
+            except (OSError, IOError) as err2:
+                _log_error("Error unlinking temporary path %s: %s", tmp_path, err2)
             raise e
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)